server/public/plugin/hooks.go

```diff
@@ -402,4 +403,11 @@ type Hooks interface {
 	//
 	// Minimum server version: 10.7
 	OnSAMLLogin(c *Context, user *model.User, assertion *saml2.AssertionInfo) error
+
+	// OnOmniSearch is invoked when a search is executed to allow the plugin to provide results to that search.
+	// It allows plugins to include their own content in the Support Packet.
+	//
+	// Minimum server version: 10.6
+	// Enterprise only
+	OnOmniSearch(c *Context, terms string, userID string, isOrSearch bool, timeZoneOffset int, page int, perPage int) ([]*model.OmniSearchResult, error)
```

Comment:
Maybe we want to have an options struct here so we can make additions if need be?